src/dataaccess/RestApiDataSource.jsx

```diff
@@ -6,14 +6,32 @@ class RestApiDataSource extends IDataSource {
     constructor() { 
         super();
         this.api = new PatientApi.DefaultApi();
+        this._gestalt = { 
+            requestTypes: {
+                async: false,
+                sync: true
+            },
+            create: false,
+            read: true,
+            update: true,
+            delete: false
+        };
     }
-    getPatient(id) {
-        // TODO: This should be used intead of the synchronous call below
-        // const callback = function(error, data, response) {
-        //     return new PatientRecord(response.text);
-        // }
-        // return this.api.getPatientById(id, callback);
-        
+
+    getGestalt() { 
+        return this._gestalt
+    }
+
+    getPatient(id, callback) {
+        // Asynchronous call
+        // return this.api.getPatientById(id, (error, data, response) => { 
```

Comment:
Should this still be commented? In the description of the task it says:
> This will also involve changing the shr-javascript-client library to remove the additional code to make a synchronous call.
> When this is working, the commented out test in DataAccess.test.js for getting a patient from the RestApiDataSource should be uncommented and should pass.

So unless something changed I think this async version should be uncommented and that test case should be uncommented and changed to use the callback method. To correctly test an async call it should look something like this where it calls `done` in the callback:
```
    it('should do a thing async', function(done) {
        dataAccess.getPatient(id, (result, error) => {
            expect(...);
            done();
        });
    });
```

Reply:
Great observation. From the discussions I had with @gregquinn2001, we would have liked to have gotten the RestApiDataSource working but we didn't want to hold up this task if we couldn't. In this case, I made changes to the shr_rest_client for testing but ran into a litany of CORS issues that were relentless. Because of that, I decided to punt on that for now, particularly because our next focus should be on getting @dehall's new data source up and running and not this older data source.

tl;dr Tried to fix this async; ran into CORS issues; punting because this dataSource isn't a priority